File: lee/src/main/java/org/example/lee/题目/链表/a_环形链表.java
package org.example.lee.题目.链表;

import org.example.lee.model.ListNode;



public class a_环形链表 {

    /**
     * has cycle
     * 快慢指针
     * @param head head
     * @return boolean
     */
    public boolean hasCycle(ListNode head) {
        ListNode low =head, fast = head;
        while (fast!=null && fast.next!=null){
            fast = fast.next.next;
            low = low.next;
            if(fast==low){
                return true;
            }
        }
        return false;
    }
}

File: lee/src/main/java/org/example/lee/题目/链表/a_环形链表.go
package 链表

func hasCycle(head *ListNode) bool {
	a, b := head, head.Next
	for a != b {
		if b == nil || b.Next == nil {
			return false
		}
		b = b.Next.Next
		a = a.Next
	}
	return true
}
